File: backend/semgrep_rules/tests/javascript/kisa_sw_16_buffer_overflow.js
function unsafe(size) {
  // ruleid: kisa.sw16.javascript.unsafe-buffer-operation-review
  return Buffer.allocUnsafe(size);
}
function safe(size) {
  // ok: kisa.sw16.javascript.unsafe-buffer-operation-review
  return Buffer.alloc(size);
}

File: backend/semgrep_rules/tests/java/kisa_sw_16_buffer_overflow.java
class kisa_sw_16_buffer_overflow {
    void unsafe(sun.misc.Unsafe u, long src, long dst, long size) {
        // ruleid: kisa.sw16.java.unsafe-native-memory-operation-review
        u.copyMemory(src, dst, size);
    }
    byte[] safe(byte[] src) {
        // ok: kisa.sw16.java.unsafe-native-memory-operation-review
        return java.util.Arrays.copyOf(src, src.length);
    }
}
